src/main/java/github/scarsz/discordsrv/listeners/DiscordAccountLinkListener.java

```diff
@@ -45,8 +48,24 @@ public void onPrivateMessageReceived(PrivateMessageReceivedEvent event) {
 
         DiscordSRV.api.callEvent(new DiscordPrivateMessageReceivedEvent(event));
 
+        // don't link accounts if config option is disabled
+        if (!DiscordSRV.config().getBoolean("MinecraftDiscordAccountLinkedUsePM")) return;
+
+        String reply = DiscordSRV.getPlugin().getAccountLinkManager().process(event.getMessage().getContentRaw(), event.getAuthor().getId());
+        if (reply != null) event.getMessage().reply(reply).queue();
+    }
+
+    @Override
+    public void onGuildMessageReceived(GuildMessageReceivedEvent event) {
+        // don't process messages sent by the bot
+        if (event.getAuthor().getId().equals(event.getJDA().getSelfUser().getId())) return;
+
+        // if message is not in the link channel, don't process it
+        TextChannel linkChannel = DiscordSRV.getPlugin().getDestinationTextChannelForGameChannelName("link");
+        if (!Objects.equals(event.getChannel(), linkChannel)) return;
```

Comment:
This should just be `!event.getChannel().equals(linkChannel)`

Reply:
I fixed it.